### ISSUE TITLE
Select - maintain current value in input as editable text

### ISSUE THREAD

**Issue by amcclain** [exhi/hoist-react#880]
The default react-select typeahead / autocomplete implementation shows the current value within the input, but it's effectively a placeholder - you can't actually edit it, even though it appears as though you should be able to do so.  (This can be verified with the first real example on their homepage.)

This can be frustrating in e.g. the following scenario:
* User searches a long list of options with variations near the end - e.g. a list of instruments with particular contracts such as `XYZ Corp Option1243.56 June 2018`
* They wish to choose the 2018 option but instead select 2019
* To get the entries again, they need to retype the entire string

See https://github.com/JedWatson/react-select/issues/1826 for a ticket in the library repo where other users also expect this behavior.  There is a workaround suggested near the end that might be promising - we should investigate. The Hoist managed wrapper gives us a good place to handle this if possible.

**Comment by lbwexler** [comment]
👍 👍 

This one is painful, agree.

**Comment by shravnk** [comment]
I caught #906 up with develop and reviewed the logic that I put in place and tested it again across `Select`'s various prop permutations. Think the question remains whether it's worth the non-trivial increase in the component's complexity.

**Comment by lbwexler** [comment]
Tom -- We think this is a high priority, and worth working on during your trip.  Can you take a look at Brendan's branch, and also the recent ticket updates in the react-select ticket above and see if you can push this over the line?

**Comment by TomTirapani** [comment]
Took a look at this - looks good.

Discussed with Brendan and removed the `controlShouldRenderValue` flag, which didn't seem to be doing much for us.

I made a couple of other minor tweaks; I made `isTextEditable` into an `editableMode` flag, to exist alongside our other "modes". Also replaced the `@bindable` on inputValue with an `@observable` and `@action` - to me, `@bindable` implies a "public" observable, whereas this is more "private".

Think this is in a good place to merge!
